src/filters.rs

```diff
@@ -390,29 +390,50 @@ pub fn modulo(input: &Value, args: &[Value]) -> FilterResult {
     }
 }
 
-pub fn escape(input: &Value, args: &[Value]) -> FilterResult {
+/// Returns the number of already escaped characters.
+fn nr_escaped(text: &str) -> usize {
+    for prefix in &["lt;", "gt;", "#39;", "quot;", "amp;"] {
+        if text.starts_with(prefix) {
+            return prefix.len();
+        }
+    }
+    0
+}
+
+// The code is adapted from
+// https://github.com/rust-lang/rust/blob/master/src/librustdoc/html/escape.rs
+// Retrieved 2016-11-19.
+fn _escape(input: &Value, args: &[Value], once_p: bool) -> FilterResult {
     try!(check_args_len(args, 0));
     match *input {
         Str(ref s) => {
-            // Adapted from
-            // https://github.com/rust-lang/rust/blob/master/src/librustdoc/html/escape.rs
-            // Retrieved 2016-11-19.
             let mut result = String::new();
             let mut last = 0;
+            let mut skip = 0;
             for (i, c) in s.chars().enumerate() {
+                if skip > 0 {
+                    skip -= 1;
+                    continue;
+                }
                 match c as char {
-                    '<' | '>' | '&' | '\'' | '"' => {
+                    '<' | '>' | '\'' | '"' | '&' => {
                         result.push_str(&s[last..i]);
+                        last = i + 1;
                         let escaped = match c as char {
                             '<' => "&lt;",
                             '>' => "&gt;",
-                            '&' => "&amp;",
                             '\'' => "&#39;",
                             '"' => "&quot;",
+                            '&' => {
+                                assert_eq!(skip, 0);
```

Comment:
The assert is harmless here. We won't reach this far if `skip` is > 0.
```
            let mut skip = 0;
            for (i, c) in s.chars().enumerate() {
                if skip > 0 {
                    skip -= 1;
                    continue;
                }
                match c as char {
[...]
```
The assert expresses our confidence in the actual value of `skip`. This value cannot be anything but 0, as explained above.

That said, if you still want me to delete `assert_eq!()`, I will. Please let me know if I should.